server/api/src/rsGeneralUpdate.cpp

```diff
@@ -11,6 +11,13 @@
 
 int
 rsGeneralUpdate( rsComm_t *rsComm, generalUpdateInp_t *generalUpdateInp ) {
+    // Deprecation messages must be handled by doing the following.
+    // The native rule engine may erase all messages in the rError array.
+    // The only way to guarantee that messages are received by the client
+    // is to add them to the rError array when the function returns.
+    const irods::at_scope_exit add_deprecated_api_warning_message{
+        [&] { addRErrorMsg(&rsComm->rError, DEPRECATED_API, "rxGeneralUpdate has been deprecated."); }};
```

Comment:
I think the "Its use should be avoided" clause would be helpful here.

Reply:
Agreed.

Reply:
Please update this change in light of our deprecation rules: https://github.com/irods/irods_docs/issues/124#issuecomment-2023055718

Reply:
Done.